README.md

```diff
@@ -1,12 +1,108 @@
 # Opencensus
 
+[![CircleCI](https://circleci.com/gh/opencensus-beam/opencensus_elixir.svg?style=svg)](https://circleci.com/gh/opencensus-beam/opencensus_elixir)
+[![Hex version badge](https://img.shields.io/hexpm/v/opencensus_elixir.svg)](https://hex.pm/packages/opencensus_elixir)
+
+Wraps some [`:opencensus`][:opencensus] capabilities for Elixir users so
+they don't have to [learn them some Erlang][LYSE] in order to get
+[OpenCensus] distributed tracing.
+
+[OpenCensus]: http://opencensus.io
+[:opencensus]: https://hex.pm/packages/opencensus
+[LYSE]: https://learnyousomeerlang.com
+
 ## Installation
 
+Add `opencensus_elixir` to your `deps` in `mix.exs`:
+
 ```elixir
 def deps do
   [
-    {:opencensus_elixir, "~> 0.2.0"}
+    {:opencensus, "~> 0.9"},
+    {:opencensus_elixir, "~> 0.3.0"}
   ]
 end
 ```
 
+## Usage
+
+Wrap your code with the
+[`Opencensus.Trace.with_child_span/3`][oce-with_child_span-3] macro to
+execute it in a fresh span:
+
+```elixir
+import Opencensus.Trace
+
+def traced_fn(arg) do
+  with_child_span "traced" do
+    :YOUR_CODE_HERE
+  end
+end
+```
+
+That's the rough equivalent of the following direct use of `:opencensus`:
+
+```elixir
+def traced_fn() do
+  try do
+    :ocp.with_child_span("traced", %{fn: :traced_fn, mod: __MODULE__})
+    {:span_ctx, trace_id, span_id, trace_options, _} = :ocp.current_span_ctx()
+
+    Logger.metadata(
+      trace_id: :io_lib.format("~32.16.0b", [trace_id]) |> List.to_string(),
+      span_id: :io_lib.format("~16.16.0b", [span_id]) |> List.to_string(),
+      trace_options: trace_options
+    )
+
+    :YOUR_CODE_HERE
+  after
+    :ocp.finish_span()
+    Logger.metadata(trace_id: nil, span_id: nil, trace_options: nil)
+  end
+end
+```
+
+The `Logger.metadata/1` work won't be necessary in an Elixir version later
+than 1.9: unifying the metadata between `:logger` and `Logger` will require
+Erlang 21 or better. If that has already happened, you'll see the metadata
+repeated at the `iex` prompt if you paste this in:
+
+```elixir
+:ocp.with_child_span("traced")
+:logger.get_process_metadata()
+Logger.metadata(()
+```
```

Comment:
That as well shouldn't bother user at all. This is implementation detail that shouldn't matter.

Reply:
How about I remove that check from the README but leave it in `logger.ex`?

Reply:
Removed.